### DISCUSSION TITLE
when i used whatsapp-web.js  pkg in nodejs and deployed on vercel it gives me error "Serverless funciton has crashed"

### DISCUSSION THREAD

**Discussion by arslanahmed777** [vercel/community#2862] [Help]
### Summary

i am using whatsapp-web.js pkg in nodejs and i deployed it on vercel 
in the documentation of whatsapp-web.js  pkg they said that to run on no GUI run a command 
"sudo apt install -y gconf-service libgbm-dev libasound2 libatk1.0-0 libc6 libcairo2 libcups2 libdbus-1-3 libexpat1 libfontconfig1 libgcc1 libgconf-2-4 libgdk-pixbuf2.0-0 libglib2.0-0 libgtk-3-0 libnspr4 libpango-1.0-0 libpangocairo-1.0-0 libstdc++6 libx11-6 libx11-xcb1 libxcb1 libxcomposite1 libxcursor1 libxdamage1 libxext6 libxfixes3 libxi6 libxrandr2 libxrender1 libxss1 libxtst6 ca-certificates fonts-liberation libappindicator1 libnss3 lsb-release xdg-utils wget"
  how can i run this command in vercel

### Example

https://testing-backend-ten.vercel.app/users

### Steps to Reproduce

plz

**Comment by smaeda-ks** [comment]
@arslanahmed777 Since this is a normal npm package, you just need to run the `npm install whatsapp-web.js` command locally in your project root, then it will update your `package.json` file.  All you need is to properly include third-party libraries in your `package.json` file. You don't need to manually install it that way.
https://www.npmjs.com/package/whatsapp-web.js/v/1.16.4-alpha.0

**Reply by arslanahmed777** [reply]
i already did this but it is giving me error

**Reply by smaeda-ks** [reply]
What errors? Please always include details as much as possible. You can also include the link to the failed deployment here

**Reply by arslanahmed777** [reply]
here is the link
https://testing-backend-ten.vercel.app/users 
here is the screen shot of my code
![image](https://github.com/vercel/community/assets/67457531/7b29937e-4311-4044-894b-b83b40297f91)

**Comment by nandasafiqalfiansyah** [comment]
same me https://testing-api-ecru.vercel.app/qr

my code link : https://github.com/nandasafiqalfiansyah/testing-api

**Comment by Giasinit** [comment]
Hi everyone!
I think the problem is from vercel.json.

When you use node.js project, you need to build your vercel project using `@now/node`.
Example vercel.json:
```
{
    "version": 2,
    "builds": [
        {
            "src": "index.js",
            "use": "@now/node"
        }
    ],
    "routes": [
        {
            "src": "/(.*)",
            "dest": "index.js"
        }
    ]
}

```

So.. the package.json doesn't work if you didn't insert
```
{
            "src": "package.json",
            "use": "vercel-sapper"
},
```

But i'm not sure..

**Reply by edu-amr** [reply]
Is it possible to host a very simple whatsapp-web.js bot on Vercel?

**Reply by edu-amr** [reply]
no it's not possible to do this, hosted on heroku, and I moved the lib to baileys because of costs and performance

**Comment by alexojo** [comment]
@arslanahmed777 could you deploy your project in project ??

**Comment by Xantino1997** [comment]
@arslanahmed777  Hello, were you able to solve the problem? The same thing happens to me no matter how I try everything.

[a comment by an automated account is not shown]
